### PR TITLE
n8n: remove systemd option incompatible with nodejs

### PR DESCRIPTION
**Pull request by purcell**
###### Motivation for this change

The `MemoryDenyWriteExecute` systemd option is [widely](https://lists.archlinux.org/pipermail/arch-commits/2021-July/1026311.html) [known](https://github.com/netdata/netdata/issues/773) to be incompatible with nodejs, and causes service crashes as reported in #119687.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by purcell** [comment]
I've confirmed that `mkForce`-ing this option to "no" fixes the related issue, so this should be the correct fix. I haven't built exactly this change on the corresponding machine though, sorry. Understand if that's a blocker, but I think there's no danger here, so submitting as-is for now.

**Comment by alyssais** [comment]
> The MemoryDenyWriteExecute systemd option is widely known to be incompatible with nodejs, and causes service crashes as reported in #119687.

Please include detail like this in your commit message as well as the PR body.  It makes it much easier to review history in the future, and will remain accessible forever in git history, whereas one day GitHub might be gone. :)

**Comment by purcell** [comment]
> > 
> 
> Please include detail like this in your commit message as well as the PR body. It makes it much easier to review history in the future, and will remain accessible forever in git history, whereas one day GitHub might be gone. :)

Yes, good point - done, thanks!

File: nixos/modules/services/misc/n8n.nix [hunk 1]
```diff
@@ -66,7 +66,6 @@ in
         RestrictNamespaces = "yes";
         RestrictRealtime = "yes";
         RestrictSUIDSGID = "yes";
```
Comment (baloo) [inline nixos/modules/services/misc/n8n.nix:68]:
```suggestion
        RestrictSUIDSGID = "yes";
        MemoryDenyWriteExecute = "no"; # v8 JIT requires memory segments to be Writable-Executable.
```

Absence of the entry is not the same as the entry being explicitly set to no, along with a comment.

**Review by baloo — CHANGES_REQUESTED** [review]

File: nixos/modules/services/misc/n8n.nix (hunk 1 above)
Comment (purcell) [reply nixos/modules/services/misc/n8n.nix:68]: Done, thanks, yes - better to be explicit.

**Comment by alyssais** [comment]
@OfBorg test n8n

**Comment by purcell** [comment]
Thank you @alyssais!

**Pull request merged**